GLOSSARY.md

```diff
@@ -1,38 +1,156 @@
+  
+## Automatic instrumentation
+Automatically adding hooks to a program. This can be achieved by a source to source transformation program, by the compiler, or by a binary instrumentation program. Automatically inserted instrumentation will typically target well-defined specific events such as function or basic block entry and exit, or memory accesses.
+
+## Binary instrumentation
+Instrumenting the compiled binary of an application, rather than the source code.
+
+## CTF: Common Trace Format
+A compact binary trace format defined by the DiaMon group of the Linux Foundation and supported by several tracing tools such as LTTng and Trace Compass.
+
+## Compute Kernel[^5]
+What is invoked by the Host to run on the GPU is called a Kernel. A Kernel is a scalar C program that defines a single thread’s execution. Prior to the threads execution the data it is to operate on is transferred to GPU. The invocation of the kernel function typically has pointers to data areas and scalars such as integers and floats that parameterize the execution. What is different from a normal C function invocation is, of course, that multiple instances of the function are invoked and each is assigned to a an individual GPU thread.
+
+## Conditional tracepoint
+A tracepoint for which, when executed, an event is generated only if a certain condition is met. This is useful because checking a condition is typically much faster than writing an event to a file.
+ 
+## Current state
+The state of all relevant resources, in the model constructed when reading a trace, at a given point in time. It is obtained by processing all the events with an earlier timestamp, and computing the state of resources accordingly.
+
+## Debugging
+Running a program through a debugger. A debugger can control the execution (set breakpoints, stop and resume the execution) and look at the content of memory, including variables and the stack.
+
+## Distributed tracing
+Tracing a distributed system. A popular API for distributed tracing is [OpenTelemetry](https://opentelemetry.io/), and many groups will equate distributed tracing to OpenTelemetry like traces.
 
-## Tracepoint[^2]
-A tracepoint is a statement placed directly in the code of an application that provides a hook to invoke a probe. A tracepoint generally provides a way in which it can be enabled or disabled dynamically.
+## Distributed trace
+A trace generated on several computers part of a distributed system.
```

Comment:
I think you need some punctuation here.
"A trace generated on several computers - part of a distributed system."?